tests/scripts/link/script.cli

```diff
@@ -0,0 +1,234 @@
+# Functional tests for project linking
+# Key:
+#   - # <comment>
+#   - $ <command>
+#   - ? <grok pattern>
+#   - ! <literal>
+#   - repeat { ... }
+#   - option { ... }
+#   - ignore { ... }
+#   - choice { ... }
+#   - unordered { ... }
+
+#pattern VERSION \d+\.\d+(?:\.\d+)?(?:-(?:alpha|beta|rc|dev)\.\d+)?\+(?:[a-f0-9]{7}|local)
+#test %{VERSION} "Gel CLI 7.2.0-dev.1306+7d4a485"
+ignore {
+    ? Newer version of gel tool exists %{VERSION} \(current %{VERSION}\). To upgrade run `gel cli upgrade`
+    ? WARNING %{GREEDYDATA} postgres: %{GREEDYDATA}
+    ? CRITICAL %{GREEDYDATA} postgres: %{GREEDYDATA}
+}
+
+$ gel instance destroy -I inst1 --force
+%ALLOW_FAILURE
+*
+
+$ gel instance destroy -I project2 --force
+%ALLOW_FAILURE
+*
+
+$ gel --version
+? Gel CLI %{VERSION}
+
+$ gel server list-versions
+? ┌%{DATA}┬%{DATA}┬%{DATA}┐
+? │%{DATA}│%{DATA}│%{DATA}│
+repeat {
+    choice {
+        ? │ stable %{DATA}
+        ? │ nightly %{DATA}
+        ? │ testing %{DATA}
+    }
+}
+? └%{DATA}┴%{DATA}┴%{DATA}┘
```

Comment:
Woah, this is very cool.

I'll have to learn how to read grok patterns. Where is `ALLOW_FAILURE` defined?
I see that there is a definition for VERSION in the comment above, but there isn't anything for DATA or GREEDYDATA.

Reply:
This is a decent list of Grok patterns: https://docs.mezmo.com/telemetry-pipelines/using-grok-to-parse

For some reason these aren't well-documented outside a few pages on the web.

I need to update this script a bit but where you see "%ALLOW_FAILURE" that's basically "%EXIT any" (ie: any exit code)